### PR TITLE
[engine] Engine cache for `VerifyCrossLink`

### PR DESCRIPTION
**Pull request by JackyWYX**
## Description

1. Added a new method `VerifyCrossLink` to engine interface.
2. Refactored some code of`engineImpl` to use the same cache for crosslink and headers.
3. Basically, `VerifyCrossLink` `VerifySeal` `VerifyHeaderSignature` will share the same cache to reduce duplicate signature verification.

## Test

tested locally.

### CODE REVIEW

File: internal/chain/engine.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 )
 
 const (
```
Comment (rlan35) [inline internal/chain/engine.go:29]: Make it 100?

**Review by rlan35 — APPROVED** [review]

**Comment by rlan35** [comment]
Looks great. Please run a mainnet node on shard 0 with this binary to see if it can verify blocks without problems

File: internal/chain/engine.go (hunk 1 above)
Comment (JackyWYX) [reply internal/chain/engine.go:29]: Sure

**Comment by JackyWYX** [comment]
> Looks great. Please run a mainnet node on shard 0 with this binary to see if it can verify blocks without problems

Run a node connecting to latest mainnet node, and crosslink verification success observed with custom print message. I think this PR is good to go.

**Pull request merged**